NAS-102047: Add ACL mode/manager docs (#1292)
freenas/freenas-docs: userguide/storage.rst

   | User              | drop-down menu   | User account to which this ACL entry applies. Only visible when *User* is the chosen :guilabel:`Tag`.      |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+
   | Group             | drop-down menu   | Group to which this ACL entry applies. Only visible when *Group* is the chosen :guilabel:`Tag`.            |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+
   | ACL Type          | drop-down menu   | How the :guilabel:`Permissions` are applied to the chosen :guilabel:`Tag`. Choose *Allow* to grant the     |
   |                   |                  | specified permissions and *Deny* to restrict the specified permissions.                                    |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Permissions Type  | drop-down menu   | Choose a list of :guilabel:`Permissions`. *Basic* shows general permissions. *Advanced* shows each         |

Reviewer: ```suggestion
   | Permissions Type  | drop-down menu   | Choose the type of permissions. *Basic* shows general permissions. *Advanced* shows each               |
```

.. _storage_acl_tab:

.. table:: ACL Options
   :class: longtable

   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+
   | Setting           | Value            | Description                                                                                                |
   |                   |                  |                                                                                                            |
   +===================+==================+============================================================================================================+
   | Path              | string           | System location of the dataset that is being modified. Read-only.                                          |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+
   | User              | drop-down menu   | Select the user to control the dataset. This user always has permissions to read or write the ACL and read |

>    |                   |                  | or write attributes. Users manually created or imported from a                                             |

Reviewer: ```suggestion
   |                   |                  | or write attributes. Users created manually or imported from a                                             |
```

   |                   |                  | :ref:`directory service <Directory Services>` appear in the drop-down menu.                                |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Group             | drop-down menu   | Select the group to control the dataset. This group always has permissions to read or write the ACL and    |

Reviewer: ```suggestion
   | Group             | drop-down menu   | The group which controls the dataset. This group always has permissions to read or write the ACL and          |
```

>    |                   |                  | read or write attributes. Groups manually created or imported from a                                       |

Reviewer: ```suggestion
   |                   |                  | read or write attributes. Groups created manually or imported from a                                       |
```

   |                   |                  | :ref:`directory service <Directory Services>` appear in the drop-down menu.                                |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Tag               | drop-down menu   | Access Control Entry (ACE) type. Select *User* or *Group* to define a specific user or group for this      |

Reviewer: ```suggestion
   | Tag               | drop-down menu   | Access Control Entry (ACE) type. Select *User* or *Group* to choose a specific user or group for this      |
```

   |                   |                  | entry, *owner@* to apply this entry to the selected :guilabel:`User`, *group@* to apply this entry to the  |
   |                   |                  | selected :guilabel:`Group`, or *everyone@* to apply this entry to all users and groups. See                |
   |                   |                  | `setfacl(1) NFSv4 ACL ENTRIES <https://www.freebsd.org/cgi/man.cgi?query=setfacl>`__.                      |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | User              | drop-down menu   | User account to which this ACL entry applies. Only visible when *User* is the chosen :guilabel:`Tag`.      |

Reviewer: Nice.

   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Group             | drop-down menu   | Group to which this ACL entry applies. Only visible when *Group* is the chosen :guilabel:`Tag`.            |

Reviewer: Nice.

   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | ACL Type          | drop-down menu   | How the :guilabel:`Permissions` are applied to the tagged account. Choose *Allow* to grant the specified   |

Reviewer: Not clear what "tagged" means here.

   |                   |                  | permissions and *Deny* to restrict the specified permissions.                                              |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Permissions Type  | drop-down menu   | Define which set of :guilabel:`Permissions` to display. *Basic* shows general, broadly applied permissions |

Reviewer: It's not defining a set, it's just selecting.  Also, "broadly applied" is not very clear.  Also, break the sentence here.
```suggestion
   | Permissions Type  | drop-down menu   | Select which set of :guilabel:`Permissions` to display. *Basic* shows general permissions.                 |
```

>    |                   |                  | and *Advanced* shows each specific type of permission for fine-grain control.                              |

Reviewer: ```suggestion
   |                   |                  | *Advanced* shows each specific type of permission for finer control.                                              |
```

   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Permissions       | drop-down menu   | Select each permission to apply for the tagged account. Permission choices change according to the chosen  |

Reviewer: "Choices change according to the chosen" is kind of confusing.  Trying to avoid some of that repetition:
"Choices vary depending on the Permission Type."

   |                   |                  | :guilabel:`Permission Type`. See the :ref:`permissions list <ACE Permissions>` for descriptions            |
   |                   |                  | of each permission.                                                                                        |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Flags Type        | drop-down menu   | Define which set of ACE inheritance :guilabel:`Flags` to display. *Basic* shows unspecific inheritance     |

Reviewer: Not define, but select.

>    |                   |                  | options and *Advanced* shows specific inheritance settings for fine-grain control.                         |

Reviewer 2: ```suggestion
   |                   |                  | options and *Advanced* shows specific inheritance settings for detailed control.                           |
```

Reviewer: As above.

   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+
   | Flags             | drop-down menu   | How this ACE is applied to newly created directories and files within the dataset. *Basic* flags enable or |

>    |                   |                  | disable ACE inheritance. *Advanced* flags allow further control of how this ACE is applied to files and    |

Reviewer: Several nonspecific "this"es.  More specific as "the":
```suggestion
   |                   |                  | disable ACE inheritance. *Advanced* flags allow further control of how the ACE is applied to files and     |
```

>    |                   |                  | directories within this dataset. See the :ref:`inheritance flags list <ACE Inheritance Flags>` for         |

Reviewer: ```suggestion
   |                   |                  | directories in the dataset. See the :ref:`inheritance flags list <ACE Inheritance Flags>` for              |
```

   |                   |                  | descriptions of *Advanced* inheritance flags.                                                              |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Apply permissions | checkbox         | Apply permissions recursively to all directories and files within the current dataset.                     |

Reviewer: s/within/in/

   | recursively       |                  |                                                                                                            |
   +-------------------+------------------+------------------------------------------------------------------------------------------------------------+

>    | Strip ACLs        | checkbox         | Setting this removes all ACLs from the current dataset. ACLs are also stripped recursively to              |

Reviewer: Change to present tense:
```suggestion
   | Strip ACLs        | checkbox         | Set to remove all ACLs from the current dataset. ACLs are also recursively stripped from                     |
```

  applied to view nonspecific directory contents.

* *Full Control* : Apply all permissions.

**Advanced Permissions**

* *Read Data* : View file contents or list directory contents.

* *Write Data* : Create new files or modify any part of a file.

* *Append Data* : Add new data to the end of a file.

* *Read Named Attributes* : view the named attributes directory.

* *Write Named Attributes* : create a named attribute directory. Must be
  paired with the *Read Named Attributes* permission.

* *Execute* : Execute a file, move through, or search a directory.

* *Delete Children* : delete files or subdirectories from inside a
  directory.

* *Read Attributes* : view file or directory non-ACL attributes.

* *Write Attributes* : change file or directory non-ACL attributes.

* *Delete* : remove the file or directory.

* *Read ACL* : view the ACL.

* *Write ACL* : change the ACL and the ACL mode.

* *Write Owner* : change the user and group owners of the file or
  directory.

* *Synchronize* : synchronous file read/write with the server.

.. _ACE Inheritance Flags:

Basic inheritance flags only enable or disable ACE inheritance. Advanced

> flags offer more fine control for applying an ACE to new files or

Reviewer 2: ```suggestion
flags offer more detailed control for applying an ACE to new files or
```

Reviewer: "finer"

directories.

* *File Inherit* : new files inherit the ACE without any inheritance
  flags.

* *Directory Inherit* : new subdirectories inherit the full ACE.

* *No Propagate Inherit* : new subdirectories inherit the ACE without any
  inheritance flags.

> * *Inherit Only* : Remove the ACE from permissions checks but allow it to

Reviewer 2: ```suggestion
* *Inherit Only* : Remove the ACE from permission checks but allow it to
```

Reviewer: Agreed.
